go/pkg/client/client.go

```diff
@@ -234,6 +237,16 @@ func (s CompressedBytestreamThreshold) Apply(c *Client) {
 	c.CompressedBytestreamThreshold = s
 }
 
+// An UploadCompressionClassifier determines what kind of compression to use for a blob on upload.
+// Note that the CompressedBytestreamThreshold takes priority over this (i.e. if the blob to be uploaded
+// is smaller than the threshold, this will not be called to classify it).
+type UploadCompressionClassifier func(*uploadinfo.Entry) repb.Compressor_Value
```

Comment:
I think this could simply return a boolean rather than a compressor. I believe the choice of the compression algorithm is mostly influenced by compute resources rather than the content itself. Another reason is that right now only ZSTD is supported by the SDK so the generalization adds complexity without immediate benefits.

Reply:
Yeah, I did wonder about this. I was aiming for a more flexible interface that wouldn't need to break compatibility in future if you did want to support other compressors. I agree that's a little theoretical though so I'll swap it to a bool.